runelite-client/src/main/java/net/runelite/client/ui/FontManager.java

```diff
@@ -24,49 +24,62 @@
  */
 package net.runelite.client.ui;
 
-import javax.swing.text.StyleContext;
+import com.google.common.collect.ImmutableBiMap;
+import java.awt.Canvas;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
 import java.awt.Font;
 import java.awt.FontFormatException;
 import java.awt.GraphicsEnvironment;
 import java.io.IOException;
+import net.runelite.client.config.FontType;
 
 public class FontManager
 {
 	private static final Font runescapeFont;
 	private static final Font runescapeSmallFont;
 	private static final Font runescapeBoldFont;
 
+	private static final ImmutableBiMap<String, Font> fontMap;
+
 	static
 	{
 		GraphicsEnvironment ge = GraphicsEnvironment.getLocalGraphicsEnvironment();
 
 		try
 		{
-			Font font = Font.createFont(Font.TRUETYPE_FONT,
+			runescapeFont = Font.createFont(Font.TRUETYPE_FONT,
 				FontManager.class.getResourceAsStream("runescape.ttf"))
 				.deriveFont(Font.PLAIN, 16);
-			ge.registerFont(font);
-
-			runescapeFont = StyleContext.getDefaultStyleContext()
-					.getFont(font.getName(), Font.PLAIN, 16);
-			ge.registerFont(runescapeFont);
 
-			Font smallFont = Font.createFont(Font.TRUETYPE_FONT,
+			runescapeSmallFont = Font.createFont(Font.TRUETYPE_FONT,
 				FontManager.class.getResourceAsStream("runescape_small.ttf"))
 				.deriveFont(Font.PLAIN, 16);
-			ge.registerFont(smallFont);
 
-			runescapeSmallFont = StyleContext.getDefaultStyleContext()
-					.getFont(smallFont.getName(), Font.PLAIN, 16);
-			ge.registerFont(runescapeSmallFont);
+			runescapeBoldFont = Font.createFont(Font.TRUETYPE_FONT,
+				FontManager.class.getResourceAsStream("runescape_bold.ttf"))
+				.deriveFont(Font.PLAIN, 16);
+
+			final LinkedHashMap<String, Font> _fontMap = new LinkedHashMap<>();
+			_fontMap.put("Runescape small", runescapeSmallFont);
```

Comment:
This map should contain only "RuneScape" font. Small/Reg/Bold is decided in getFontOfType